File: frontend/src/types/Bot.ts
export const botLanguages = ['python', 'cpp', 'typescript'] as const;
export type BotLanguage = typeof botLanguages[number];

export type Bot = {
    id: string;
    name: string;
    language: BotLanguage;
    rating: number;
    userId: string;
    creationDate: string;
};

export type BotCreate = {
    name: string;
    language: BotLanguage;
}

File: backend/schema/bot.py
from datetime import datetime

from schema.bot_language import BotLanguage
from schema.base_schema import BaseSchema

class Bot(BaseSchema):
    id: str
    name: str
    language: BotLanguage
    rating: float
    user_id: str
    creation_date: datetime

class BotCreate(BaseSchema):
    name: str
    language: BotLanguage